invenio_curations/services/diff.py

```diff
@@ -0,0 +1,252 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 Graz University of Technology.
+#
+# Invenio-Curations is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Diff handling module."""
+from abc import ABC, abstractmethod
+from typing import Any, Final, cast
+
+from flask import current_app, render_template
+from invenio_i18n import lazy_gettext as _
+
+from .utils import HTMLParseError, cleanup_html_tags
+
+_DIFF_TYPE = tuple[str, str, list | tuple]
+
+
+class DiffError(Exception):
+    """Custom exception for diff exceptions."""
+
+
+class DiffBase(ABC):
+    """Interface for classes that process diffs in context of comment creation/update."""
+
+    @abstractmethod
+    def validate_and_cleanup(self) -> Any:
+        """Validate the current diff, and remove unwanted data."""
+        ...
```

Comment:
are the ellipsis needed? i thought if the docstring exists the ellipsis is not necessary